features/foutafhandeling.feature

```diff
@@ -0,0 +1,542 @@
+
+Rule: wanneer een parameter geometrie bevat in een crs anders dan RD (epsg:28992) wordt een 400 fout gegeven
+
+    Abstract Scenario: accept-crs WGS84 geeft een foutmelding
+        Als '<path><query string>' request wordt gedaan met header 'Accept-Crs: epsg:4326'
+        Dan bevat de response ten minste de volgende velden
+        | naam   | waarde                                                              |
+        | title  | Coördinatenstelsel epsg:4326 in Content-Crs wordt niet ondersteund. |
+        | status | 400                                                                 |
+        | code   | crsNotSupported                                                     |
+        En bevat de response geen veld "invalidParams"
+        En is de http status code van het antwoord 400
+        En bevat de response de header "Content-Type" met waarde "application/problem+json"
+
+        Voorbeelden:
+        | path                   | query string                            |
+        | /panden                | ?locatie=98095,438495                   |
+        | /panden                | ?locatie=51.93201,4.55980               |
+        | /panden                | ?bbox=135228,457502,135246,457457       |
+        | /adresseerbareobjecten | ?bbox=135228,457502,135246,457457       |
+        | /adresseerbareobjecten | ?bbox=52.10540,5.09861,52.10500,5.09887 |
+
+    Abstract Scenario: zoekgeometrie die niet klopt met de opgegeven of default crs geeft een foutmelding
+        Als '<path><query string>' wordt aangeroepen
+        Dan bevat de response ten minste de volgende velden
+        | naam   | waarde                                        |
+        | title  | Een of meerdere parameters zijn niet correct. |
+        | status | 400                                           |
+        | code   | paramsValidation                              |
+        En bevat de response de volgende invalidParams
+        | name        | code             | reason                                |
+        | <parameter> | geometryMismatch | Waarde is niet conform opgegeven CRS. |
+        En is de http status code van het antwoord 400
+        En bevat de response de header "Content-Type" met waarde "application/problem+json"
+
+        Voorbeelden:
+        | path                   | query string                            | parameter |
+        | /panden                | ?locatie=51.93201,4.55980               | locatie   |
+        | /panden                | ?bbox=52.10540,5.09861,52.10500,5.09887 | bbox      |
+        | /adresseerbareobjecten | ?bbox=52.10540,5.09861,52.10500,5.09887 | bbox      |
+
+Rule: wanneer er geen API key wordt gestuurd wordt een 400 fout gegeven met html response
+
+    Scenario: geen api key in request
+        Als '/adressen/0599200000193766' request wordt gedaan zonder header 'x-api-key'
```

Comment:
In postman krijgt je inderdaad deze header en tekst. Dit komt omdat de Accept header de waarde */* heeft. Als je deze aanpast naar application/json (wat axios doet als http client), dan krijgt je json response. De laatste twee dan stappen kunnen dan naar het volgende worden aangepast
```
        En bevat de response de header "Content-Type" met waarde "application/json;charset=utf-8"
        En bevat de response ten minste de volgende velden
        | naam             | waarde                                |
        | error            | Kadaster - Foute aanvraag.            |
        | errorDescription | Het bericht kan niet worden verwerkt. |
        | errorDetail      | Missing API Key                       |
```

Reply:
we weten niet zeker of elke client accept: Application/json stuurt, toch?

Kunnen we het niet beperken tot:
```
        Dan is de http status code van het antwoord 401
        En bevat de responsebody de tekst "Invalid API Key"
```

Reply:
Tja, in dat geval zou ik dan ook de stap `En bevat de responsebody de tekst "Missing API Key"` weglaten. Dit is zo breed dat een consumer developer niets mee kan.